tests/integration/packet/packet.exp

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env -S expect -f
+#
+# This is an expect script modeling the user interaction of PATE on the packet example.
+#
+# Beyond testing basic verifier functionality, this test is also intended to
+# provide an integration test of the 1) terminal user interaction, and 2)
+# resulting equivalence condition, in ways that the other automated tests do not.
+# The test for the "expected" equivalence condition is syntactic, so this test will
+# likely be more brittle (e.g. break if the term simplification changes such that
+# intermediate variable numbers change) and may need to be updated to reflect
+# intended subsequent changes to PATE verification details.
+#
+# This test is intended to be run from the root of the pate directory:
+#    ./tests/integration/packet/packet.exp
+#
+# Note that expected strings must be escaped: double-quotes (\") and brackets (\\\[)
+
+spawn ./pate.sh --original tests/integration/packet/exe/packet.exe --patched tests/integration/packet/exe/packet.patched.exe -s parse_packet
+
+set timeout 480
+
+expect_before {
+        timeout { send_user "\n\nFAILURE: timeout\n"; exit 1 }
+        eof     { send_user "\n\nFAILURE: eof\n";     exit 2 }
+}
+
+expect "Choose Entry Point"
+expect "1: Function Entry \"parse_packet\" (segment1+0x554)"
```

Comment:
One of the main motivations for the scripting interface was to avoid referencing specific numbers from the menus. I'm assuming that's not something we can easily do here?

Reply:
It's definitely an option to, for example, regex parse the lines of menus from PATE to get the appropriate number to send in response to each step. That would reduce some of the brittleness due to menu item ordering, but we'd still have either some level of brittleness or some level of ambiguity matching the rest of each line depending on how strictly we check for things.

My thinking here is that we have the rest of the PATE scripts which do the more structural matching for the other test cases - for this integration test, given that it'll still be brittle in ways beyond just the menu numbers, we might prioritize keeping the expect script as simple as possible and just use the basic string match. Doing it this way means that if/when we need to make updates to the expect script, it's just a matter of copy-pasting in the expected line, rather than adapting or writing a new regex expression to capture the appropriate context. I think that's an ok tradeoff for this single small example - if this ends up flagging too many issues that *aren't* interesting to us, or if we later end up doing lots more / longer integration tests like this, then I think we may want to explore something that does more sophisticated parsing of PATE output.